src/System.IO.Compression/src/System/IO/Compression/Crc32Helper.cs

```diff
@@ -7,12 +7,16 @@
 // IEEE TRANSACTIONS ON COMPUTERS, VOL. 57, NO. 11, NOVEMBER 2008
 
 using System.Diagnostics;
+using System.Security;
 
 namespace System.IO.Compression
 {
     internal static class Crc32Helper
     {
-        // Generated tables for crc calculation.
+        private enum WorkerType : byte { Unknown = 0, Managed = 1, ZLib = 2 };
+        private static readonly WorkerType s_CrcType = GetCrcType();
```

Comment:
I realize you're just copying the WorkerType present elsewhere in the implementation.  But, FWIW, I think that solution is unnecessarily complicated, and since this is standalone, we don't need to copy it here.  How about just having:
```
private static readonly bool s_useZlib = IsZLibCrcAvailable();
...
private static bool IsZLibCrcAvailable() { ... }
...
return s_useZLib ?
    crc32_zlib(...) :
    crc32_managed(...);
```